contracts/Wrapper.sol

```diff
@@ -32,53 +37,69 @@ contract Wrapper is PartialCommonOwnership721 {
     )
   {}
 
+  /// @notice helper function to get wrapperTokenId: hash the original NFT
+  ///         contract address and its respective local tokenId
+  /// @param assetContract Address the contract defining the token
+  /// @param tokenId Token Id to be wrapped
+  function createWrappedTokenId(
+    address assetContract, 
+    uint256 tokenId
+  ) public returns (uint256) {
+    return uint256(uint32(bytes4(keccak256(abi.encode(assetContract, tokenId)))));
+  }
+
+  /// @notice
+  /// @param operator Address of contract TODO
+  /// @param from Address the owner of the token
+  /// @param tokenId Token Id to be wrapped
+  /// @param _data extra data - unused
+  function onERC721Received(
+    address operator, 
+    address from, 
+    uint256 tokenId, 
+    bytes memory _data
+  ) public returns (bytes4) {
+    uint256 wrapperTokenId = createWrappedTokenId(msg.sender, tokenId);
+    _safeMint(from, wrapperTokenId);
+    return bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"));
+  }
+
   /// @notice
   /// @param _tokenContractAddress Address of contract that issued token
   /// @param _tokenId Token Id to be wrapped
-  /// @param _beneficiary Note: requires support for multiple
-  /// @param _taxRate Note: requires support for multiple
-  /// @param _taxationPeriod Note: requires support for multiple
-  function deposit(
-    address _tokenContractAddress,
-    uint256 _tokenId,
-    address payable _beneficiary,
```

Comment:
I'm going to work towards getting the multiples support in by EOD so these params can be maintained.

Reply:
Multi- beneficiary, tax rate and tax period support has been merged in.

Reply:
I saw the PRs/patches, makes sense!

Unfortunately I'm not used to doing merges to branches different than main so it was a bit of a puzzle for me - also being unfamiliar with the codebase - to work out what the various diffs meant.

At the moment I rebased my branch `vicsn/token-wrapper` both on `origin/token-wrapper` and on `origin/main` (origin being 721labs).

Would you like to rebase `origin/token-wrapper` on `origin/main` so this PR won't show any conflicts any more? I could do it to but before overwriting more git history I better involve you. :)